app/src/main/java/org/greenstand/android/TreeTracker/permission/Permissions.kt

```diff
@@ -0,0 +1,174 @@
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            })
+                    }
+
+                }
+            }
+            Manifest.permission.ACCESS_FINE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        // TODO Check if location is enabled from the LocationUpdateManager before calling enableLocation
+                        return
+                    }
+                    (perm.shouldShowRationale || !perm.permissionRequested) -> {
+                        AlertDialog(
+                            onDismissRequest = { navController.popBackStack() },
+                            title = {
+                                Text(text = stringResource(R.string.accept_location_permission_header)
+                                )
+                            },
+                            text = {
+                                Text(
+                                    text = stringResource(R.string.accept_location_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            })
+                    }
+
+                }
+            }
+            Manifest.permission.ACCESS_COARSE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        // TODO Check if location is enabled from the LocationUpdateManager before calling enableLocation
+                        return
+                    }
+                    (perm.shouldShowRationale || !perm.permissionRequested) -> {

```

Comment:
Looks like we handle the first time we'd show the rationale. However, if they deny the permission, they are stuck with it disabled. The dialog will says "accept permission" but nothing will happen.

To fix this I think we need to check for 'perm.permissionRequested'. If it's true, then we need to send them to the settings screen to enable it.

This applies to all the permissions we need.

Reply:
I think I get this now, I would make a fix to this.